R/comorbidity.R

```diff
@@ -84,18 +87,88 @@
 #' * `drug`, for drug abuse;
 #' * `psycho`, for psychoses;
 #' * `depre`, for depression;
+#' 
+#' For the M3 score the dataset contains the following variables:
+#' * The `id` variable as defined by the user;
+#' * `ami`, for Myocardial infarction;
+#' * `chf`, for Congestive heart failure;
+#' * `pvd`, for Peripheral vascular disease;
+#' * `aneur`, for Aortic and other aneurysms;
+#' * `ven`, for Venous insufficiency;
+#' * `cevd`, for Cerebrovascular disease;
+#' * `dementia`, for Dementia;
+#' * `bd`, for Mental and behavioural disorders due to brain damage;
+#' * `copd`, for Chronic obstructive pulmonary disease;
+#' * `conntiss`, for Connective tissue;
+#' * `pud`, for Peptic ulcer disease;
+#' * `diabunc`, for Diabetes, uncomplicated;
+#' * `diabc`, for Diabetes, complicated;
+#' * `para`, for Paralysis;
+#' * `rend`, for Renal disease;
+#' * `canccolrec`, for Colorectal cancer;
+#' * `cancbreast`, for Breast cancer;
+#' * `cancprost`, for Prostate cancer;
+#' * `canclung`, for Lung cancer;
+#' * `canclymphleuk`, for Lymphomas and leukaemias;
+#' * `cancuppergi`, for Upper gastrointestinal cancers;
+#' * `cancmela`, for Malignant melanoma;
+#' * `cancgyn`, for Gynaecological cancers;
+#' * `cancoth`, for Other cancers;
+#' * `metacanc`, for Metastatic cancer;
+#' * `msld`, for Liver disease, moderate or severe;
+#' * `aids`, for AIDS/HIV;
+#' * `ang`, for Angina;
+#' * `hypunc`, for Hypertension, uncomplicated;
+#' * `carit`, for Cardiac arrhythmias;
+#' * `pcd`, for Pulmonary circulation disorders;
+#' * `valv`, for Valvular disease;
+#' * `bdi`, for Bowel disease inflammatory;
+#' * `ond`, for Other neurological disorders exc epilepsy;
+#' * `epi`, for Epilepsy;
+#' * `mpnd`, for Muscular peripheral nerve disorder;
+#' * `mpd`, for Major psychiatric disorder;
+#' * `anxbd`, for Anxiety and Behavioural disorders;
+#' * `blood`, for Coagulopathy and other blood disorder;
+#' * `dane`, for Deficiency anaemia;
+#' * `obes`, for Obesity;
+#' * `alcohol`, for Alcohol abuse;
+#' * `drug`, for Drug abuse;
+#' * `panc`, for Pancreatitis;
+#' * `endo`, for Endocrine disorder;
+#' * `utc`, for Urinary tract problem chronic;
+#' * `tub`, for Tuberculosis;
+#' * `bone`, for Bone disorders;
+#' * `osteounc`, for Osteoporosis, uncomplicated;
+#' * `immsys`, for Immune system disorder;
+#' * `metab`, for Metabolic disorder;
+#' * `mentret`, for Mental retardation;
+#' * `cvhep`, for Chronic viral hepatitis;
+#' * `sleep`, for Sleep disorder;
+#' * `inear`, for Inner ear disorder;
+#' * `cinfnos`, for Chronic infection NOS;
+#' * `maln`, for Malnutrition nutritional;
+#' * `ceye`, for Chronic eye problem;
+#' * `cdnos`, for Cardiac disease NOS;
+#' * `intest`, for Intestinal disorder;
+#' * `jsd`, for Joint spinal disorder;
+#' * `flag_comp_diab`, for flagging for diabetes complications;
+#' * `flag_exc_osteo`, for flagging for osteoporosis exclusions;
+#' * `flag_exc_hyp`, for flagging for hypertension exclusions;
```

Comment:
If I understand correctly, these are flags to set these comorbidities to zero if any of these codes are found. Is that right? 

If so, do you think it would be a good idea to drop them before returning the dataset with comorbidities?

Reply:
Thanks Alessandro. I do think that would result in a cleaner user experience, the potential issues that I saw were:

- That is not implemented (or necessary) for the other scales, so wasn't sure if that's a direction you wanted to go.
- My understanding would be that it is implemented in `.assign0()`, but given that is called in both `comorbidity()` and `score()`, could removing columns at one stage or the other result in an error state (particularly when multiplying by weights) that requires M3-specific code in those functions?

Reply:
Thanks for the updates and clarifications.

I was exactly thinking about including this in `assign0()`, but then realised (as you point out here) that such columns would be needed by the `score()` function.

I need to think a bit about how to implement this effectively within the current infrastructure. I will leave this hanging for now, but I'll get back to you - hope that's okay!

Reply:
Of course, will wait to hear from you.